enaml/src/subscription_observer.cpp

```diff
@@ -0,0 +1,335 @@
+/*-----------------------------------------------------------------------------
+ * | Copyright (c) 2025, Nucleic Development Team.
+ * |
+ * | Distributed under the terms of the Modified BSD License.
+ * |
+ * | The full license is in the file LICENSE, distributed with this software.
+ * |----------------------------------------------------------------------------*/
+#include <iostream>
+#include <sstream>
+#include <cppy/cppy.h>
+
+#ifdef __clang__
+#pragma clang diagnostic ignored "-Wdeprecated-writable-strings"
+#endif
+
+#ifdef __GNUC__
+#pragma GCC diagnostic ignored "-Wwrite-strings"
+#endif
+
+
+namespace enaml
+{
+
+// POD struct - all member fields are considered private
+struct SubscriptionObserver
+{
+    PyObject_HEAD
+    PyObject* atomref;
+    PyObject* name;
+
+    static PyType_Spec TypeObject_Spec;
+
+    static PyTypeObject* TypeObject;
+
+    static bool Ready();
+
+    static bool TypeCheck( PyObject* ob );
+
+};
+
+
+namespace
+{
+
+
+PyObject*
+SubscriptionObserver_new( PyTypeObject* type, PyObject* args, PyObject* kwargs )
+{
+    PyObject* owner;
+    PyObject* name;
+    static char* kwlist[] = { "owner", "name", 0 };
+    if( !PyArg_ParseTupleAndKeywords( args, kwargs, "OU", kwlist, &owner, &name ) )
+    {
+        return 0;
+    }
+    cppy::ptr ptr( PyType_GenericNew( type, args, kwargs ) );
+    if( !ptr )
+    {
+        return 0;
+    }
+    SubscriptionObserver* self = reinterpret_cast<SubscriptionObserver*>( ptr.get() );
+
+    cppy::ptr atom_api( PyImport_ImportModule("atom.api") );
+    if ( !atom_api )
+    {
+        PyErr_SetString( PyExc_ImportError, "Could not import atom.api" );
+        return 0;
+    }
+    cppy::ptr atomref( atom_api.getattr("atomref") );
+    if ( !atomref )
+    {
+        PyErr_SetString( PyExc_ImportError, "Could not import atom.api.atomref" );
+        return 0;
+    }
+
+    self->atomref = PyObject_CallOneArg(atomref.get(), owner);
+    if( !self->atomref )
+    {
+        return 0;
+    }
+    self->name =  cppy::incref( name );
+    return ptr.release();
+}
+
+
+void
+SubscriptionObserver_clear( SubscriptionObserver* self )
+{
+    Py_CLEAR( self->atomref );
+    Py_CLEAR( self->name );
+}
+
+
+int
+SubscriptionObserver_traverse( SubscriptionObserver* self, visitproc visit, void* arg )
+{
+    Py_VISIT( self->atomref );
+    Py_VISIT( self->name );
+    #if PY_VERSION_HEX >= 0x03090000
+    // This was not needed before Python 3.9 (Python issue 35810 and 40217)
+    Py_VISIT(Py_TYPE(self));
+    #endif
+    return 0;
+}
+
+
+void
+SubscriptionObserver_dealloc( SubscriptionObserver* self )
+{
+    PyObject_GC_UnTrack( self );
+    SubscriptionObserver_clear( self );
+    Py_TYPE(self)->tp_free( reinterpret_cast<PyObject*>( self ) );
+}
+
+
+int
+SubscriptionObserver__bool__( SubscriptionObserver* self )
+{
+    cppy::ptr atomrefptr( cppy::incref( self->atomref ) );
+    return atomrefptr.is_truthy();
```

Comment:
We can avoid playing with ref counts here:
```suggestion
    return PyObject_IsTrue( self->atomref );
```